### PR TITLE
custom element reactions stack

### PR DESCRIPTION
**Pull request by bicknellr**
https://html.spec.whatwg.org/#custom-element-reactions-stack

Closes #91

### CODE REVIEW

File: src/Patch/Element.js [hunk 1]
```diff
@@ -117,22 +107,26 @@ export default function(internals) {
...
-     * @this {Element}
-     * @param {string} name
-     * @param {string} newValue
-     */
-    function(name, newValue) {
-      // Fast path for non-custom elements.
-      if (this.__CE_state !== CEState.custom) {
-        return Native.Element_setAttribute.call(this, name, newValue);
-      }
+      /**
```
Comment (bicknellr) [inline src/Patch/Element.js]: This continued statement has extra indentation.

File: src/CustomElementReactionsStack.js [hunk 2]
```diff
@@ -0,0 +1,130 @@
...
+   *
+   * https://html.spec.whatwg.org/#cereactions
+   */
+  pushQueue() {
+    this._queueStart[this._queueCount++] = this._length;
+  }
+
+  /**
+   * Pops and flushes the top-most queue of the reactions stack. Must be called
+   * when entering any API marked with `CEReactions`.
```
Comment (bicknellr) [inline src/CustomElementReactionsStack.js]: s/entering/exiting/

File: src/Patch/Element.js (hunk 1 above)
Comment (bicknellr) [inline src/Patch/Element.js]: fixed

File: src/CustomElementReactionsStack.js (hunk 2 above)
Comment (bicknellr) [inline src/CustomElementReactionsStack.js]: fixed

**Comment by bicknellr** [comment]
Hmm, I think some of the automatic parts of the merge didn't work out.

**Comment by bicknellr** [comment]
Ok, the problem was that before this merge there were (a) no tests in this branch for ChildNode / ParentNode and (b) neither of the methods of these interfaces had the appropriate calls to `pushCEReactionsQueue`/`popCEReactionsQueue`: the branch was broken and appeared to be passing and merging caused the new tests to fail locally. I've added the calls to push / pop queues in these methods and all the newly merged tests are now passing.

**Review by TimvdLippe — APPROVED** [review]
Statically looks good. I am not experienced with the spec, so someone else should also do a careful review. @bicknellr thanks for reviving this PR!

**Comment by dfreedm** [comment]
Remaking PR in monorepo!

**Pull request closed without merging**